scico/array.py

```diff
@@ -0,0 +1,264 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020-2022 by SCICO Developers
+# All rights reserved. BSD 3-clause License.
+# This file is part of the SCICO package. Details of the copyright and
+# user license can be found in the 'LICENSE' file distributed with the
+# package.
+
+"""Utility functions for arrays, array shapes, array indexing, etc."""
+
+
+from __future__ import annotations
+
+import warnings
+from typing import Any, List, Optional, Tuple, Union
+
+import numpy as np
+
+import jax
+from jax.interpreters.pxla import ShardedDeviceArray
+from jax.interpreters.xla import DeviceArray
+
+import scico.blockarray
+import scico.numpy as snp
+from scico.typing import ArrayIndex, Axes, AxisIndex, JaxArray, Shape
+
+__author__ = """\n""".join(
+    [
+        "Brendt Wohlberg <brendt@ieee.org>",
+        "Luke Pfister <luke.pfister@gmail.com>",
+        "Thilo Balke <thilo.balke@gmail.com>",
```

Comment:
I think I wrote "parse_axes". BTW, how careful are we trying to be with author lists?

Reply:
Sorry for the omission; now fixed. In my defense, you weren't listed in `scico.util` either. On the broader question, including this does seem to be standard, but I'm not sure what the point is, and they're a pain to maintain, so I think we should consider removing them.

Reply:
Sorry for the omission - fixed now. In my defense, you weren't listed in `scico.util` either. On the broader question, I don't really see the point, and think we should consider removing them.